schema.go

```diff
@@ -106,3 +113,69 @@ func findBuildByBuilder(db *mgo.Database, builder string, buildnum int) (*LogKee
 	}
 	return build, nil
 }
+
+func findAndDeleteTests(db *mgo.Database) ([]bson.ObjectId, error) {
+	now := time.Now()
+	query := bson.M{
+		"started": bson.M{"$lte": now.Add(-deletePassedTestCutoff)},
+		"failed":  false,
+	}
+	var testIDs []bson.ObjectId
+	err := db.C("tests").Find(query).Sort("started").Limit(maxTests).Distinct("_id", &testIDs)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding tests")
+	}
+	filter := bson.M{
+		"_id": bson.M{"$in": testIDs},
+	}
+	_, err = db.C("tests").RemoveAll(filter)
```

Comment:
I think using your original query here might be faster, because they'll all be cached and also we don't need to do a big $in